CartoDB/cartodb
# Icon styling [DON'T DEPLOY UNTIL 13/12]

This PR enables icon styling through the fill component. 

This feature is hidden from users using the `icon-styling` feature flag.

Although there are many changes in several files, most of the changes are simply passing the `modals` and `userModel` objects to the fill component, so we can launch a modal and check the feature flag inside it.

### Pending stuff

In order to speed up the development of this feature, and since it's protected under a feature flag, there are some minor details that this PR is not addressing at this moment (they'll be taken care on separate issues):

1. Small sets of specs (they cover the bare minimum, but we'll add some more in another ticket)
2. The color of the icon in the input UI gets lost after the first change of color.
3. CSS misalignment in the icon carousel.

### How to test this feature

1. Enable the `icon-styling` feature flag in your user.
2. Create a map with some geometries.
3. Go to the style tab.
4. Change the `fill` property of the geometries using the fill component.
5. Select an icon from the list or the modal.

Once you've done that, your cartoCSS should be updated with a `marker-file` property. Please, also test that disabling the feature flag hides the UI from the user.

You can have a look the UI in this issue: https://github.com/CartoDB/cartodb/issues/10452 and on [this one](https://github.com/CartoDB/design/issues/303), but bear in mind that we are not implementing file upload or styling categories just yet. 

### Deploying the feature

We'll deploy this feature and enable it for our own CARTO users first. In a next phase we'll add more  functionality and make it available for the general public. Ask @xavijam or @javierarce if you have any doubt with this.

Closes #10746
